lib/libutee/arch/arm/ftrace/ftrace.c

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2019, Linaro Limited
+ */
+
+#include <setjmp.h>
+#include <user_ta_header.h>
+#include <utee_syscalls.h>
+#include "ftrace.h"
+
+static const char hex_str[] = "0123456789abcdef";
+static uint32_t ret_idx_setjmp;
+
+static void __noprof ftrace_memmove(struct ftrace_buf *fbuf, size_t size)
+{
+	char *dst = (char *)fbuf + fbuf->buf_off;
+	const char *src = (char *)fbuf + fbuf->buf_off + size;
+	size_t len = 0;
+
+	fbuf->curr_size -= size;
+	len = fbuf->curr_size;
+
+	while (len--)
```

Comment:
While this is correct the style isn't what we're normally using. This looks like some old fashioned K&R code. I would have preferred writing this as:
``` C
for (n = 0; n < fbuf->curr_size; n++)
    dst[n] = src[n];
```
The point with using the same style is that it's easy to read as you get used to it. If different styles are used more effort is needed to understand the code. We don't change imported source code to follow any particular style, but newly written code which we review carefully should try to stick to the established style.

Reply:
Make sense, will update the style.